src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,460 @@
+
+:requirement: passkey
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.builtwith(client="passkey")
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.Client)
+def test_passkey__register__sssctl(client: Client, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a key with sssctl
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key.
+        2. Check the output.
+    :expectedresults:
+        1. New key is registered.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = client.sssctl.passkey_register(
+        username="user1",
+        domain="ldap.test",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script",
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with sssctl"
+
+
+@pytest.mark.builtwith(client="passkey")
+@pytest.mark.builtwith(KnownTopology.IPA, ipa="passkey")
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a passkey with the IPA command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup.
+    :steps:
+        1. Use ipa command to register a FIDO2 key.
+        2. Check the output that contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = (
+        ipa.user("user1")
+        .add()
+        .passkey_add_register(
+            pin=123456,
+            device=f"{moduledatadir}/umockdev.device",
+            ioctl=f"{moduledatadir}/umockdev.ioctl",
+            script=f"{testdatadir}/umockdev.script",
+        )
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with the IPA command"
+
+
+@pytest.mark.builtwith(client="passkey")
+@pytest.mark.builtwith(KnownTopology.LDAP, ldap="passkey")
+@pytest.mark.builtwith(KnownTopology.IPA, ipa="passkey")
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication of user with LDAP, IPA, AD and Samba
```

Comment:
In line with some recent feedback in another group to be more specific about terminology, I think authentication here should be listed as:   "Check su authentication of user" or "Check authentication of user with su for LDAP, IPA, AD and Samba".